tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/AzurePipelines/BlobUploadProcessor.cs

```diff
@@ -810,6 +807,12 @@ public static int CalculateBatches(int startingNumber, int batchSize = ApiBatchS
             }
         }
 
+        public async Task<bool> BuildBlobExistsAsync(Build build)
```

Comment:
As we talked it might be worth doing a blob list and checking locally instead of querying once per blob.

Reply:
We now pull a list of known blob names that overlap the period we're processing, then check to see if each blob is in the list